pkg/inject/utils.go

```diff
@@ -80,3 +80,33 @@ func containsEnvoyContainer(pod *corev1.Pod) (bool, int) {
 	}
 	return false, -1
 }
+
+func isSDSDisabled(pod *corev1.Pod) bool {
+	if v, ok := pod.ObjectMeta.Annotations[AppMeshSDSAnnotation]; ok {
+		if v == "disabled" {
+			return true
+		}
+	}
+	return false
+}
+
+func mutateSDSMounts(pod *corev1.Pod, envoyContainer *corev1.Container, SdsUdsPath string) {
+	SDSVolumeType := corev1.HostPathSocket
+	volume := corev1.Volume{
+		Name: "sds-socket-volume",
+		VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{
+				Path: SdsUdsPath,
+				Type: &SDSVolumeType,
+			},
+		},
+	}
+
+	volumeMount := corev1.VolumeMount{
+		Name:      "sds-socket-volume",
+		MountPath: SdsUdsPath,
+	}
+
+	envoyContainer.VolumeMounts = append(envoyContainer.VolumeMounts, volumeMount)
+	pod.Spec.Volumes = append(pod.Spec.Volumes, volume)
```

Comment:
To make sure this is idempotent, if a volume is already there, we should not append it again

Reply:
Agree, for VirtualGateway scenario. We can do a name based match to make sure we don't already have an existing volume.